### DISCUSSION TITLE
FMU-Python dependencies

### DISCUSSION THREAD

**Discussion by AishwaryaRaghavendra** [NTNU-IHB/PythonFMU#190] [Q&A]
Conside source system where the FMU was built from python script including all their dependencies which are thrid-party packages eg. numpy, pandas etc. using requirements.txt file.

Does this FMU when used in another destination system requires all these dependencies to be again installed in destination system as well in order to make this FMU run there?

**Comment by markaren** [comment]
Yes

**Comment by joemar25** [comment]
Yes, the FMU (Functional Mock-up Unit) created from a Python script with third-party dependencies such as numpy, pandas, etc., does require these dependencies to be installed in the destination system where the FMU is deployed.

When you build an FMU from Python code with third-party dependencies, PythonFMU does not package a full environment within the FMU. However, you can package a `requirements.txt` or `environment.yml` file within your FMU that defines your dependencies. 

When deploying the FMU, the end users will need to update their local Python environment. They can do this by executing `pythonfmu deploy -f my.fmu`, which will look in the `resources` folder for one of the following files: `requirements.txt` or `environment.yml`. This will ensure that the necessary dependencies are installed in the Python environment where the FMU will be executed. 


https://pypi.org/project/pythonfmu/
https://fmi-standard.org/docs/3.0/.
